lib/ui/hooks.dart

```diff
@@ -24,50 +24,121 @@ void _updateWindowMetrics(double devicePixelRatio,
     .._physicalSize = new Size(width, height)
     .._padding = new WindowPadding._(
       top: top, right: right, bottom: bottom, left: left);
-  if (window.onMetricsChanged != null)
-    window.onMetricsChanged();
+  _invoke(window.onMetricsChanged, window._onMetricsChangedZone);
 }
 
 void _updateLocale(String languageCode, String countryCode) {
   window._locale = new Locale(languageCode, countryCode);
-  if (window.onLocaleChanged != null)
-    window.onLocaleChanged();
+  _invoke(window.onLocaleChanged, window._onLocaleChangedZone);
 }
 
 void _updateSemanticsEnabled(bool enabled) {
   window._semanticsEnabled = enabled;
-  if (window.onSemanticsEnabledChanged != null)
-    window.onSemanticsEnabledChanged();
+  _invoke(window.onSemanticsEnabledChanged, window._onSemanticsEnabledChangedZone);
 }
 
 void _dispatchPlatformMessage(String name, ByteData data, int responseId) {
   if (window.onPlatformMessage != null) {
-    window.onPlatformMessage(name, data, (ByteData responseData) {
-      window._respondToPlatformMessage(responseId, responseData);
-    });
+    _invoke3<String, ByteData, PlatformMessageResponseCallback>(
+      window.onPlatformMessage,
+      window._onPlatformMessageZone,
+      name,
+      data,
+      (ByteData responseData) {
+        window._respondToPlatformMessage(responseId, responseData);
+      },
+    );
   } else {
     window._respondToPlatformMessage(responseId, null);
   }
 }
 
 void _dispatchPointerDataPacket(ByteData packet) {
   if (window.onPointerDataPacket != null)
-    window.onPointerDataPacket(_unpackPointerDataPacket(packet));
+    _invokeOnPointerDataPacket(_unpackPointerDataPacket(packet));
+}
+
+void _invokeOnPointerDataPacket(PointerDataPacket packet) {
+  _invoke1<PointerDataPacket>(window.onPointerDataPacket, window._onPointerDataPacketZone, packet);
```

Comment:
Because otherwise in the test I'd have to create a `ByteData` blob that has to correctly parse into a `PointerDataPacket` just to satisfy `_unpackPointerDataPacket`. This would be superfluous because the test callback ignores the value.